src/HttpClientFactory/Polly/src/HttpRequestMessageExtensions.cs

```diff
@@ -13,8 +13,8 @@ namespace Polly
     /// </summary>
     public static class HttpRequestMessageExtensions
     {
-        internal static readonly string PolicyExecutionContextKey = "PolicyExecutionContext";
-        
+        internal const string PolicyExecutionContextKey = "PolicyExecutionContext";
```

Comment:
This looks like a case where `static readonly` is better because it lets ReferenceEquals work.